### PR TITLE
DAGs: migration fro  requests lib to http hook

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: dags/library/utils.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
 def get_url(key, year):
     url = {
-        "publications_total_count": r"http://cdsweb.cern.ch/search?wl=0&ln=en&cc=Published+"
+        "publications_total_count": r"search?wl=0&ln=en&cc=Published+"
```
Comment (drjova) [inline dags/library/utils.py:3]: these are not urls any more, these are endpoints :)

File: dags/library/cern_publication_records.py [hunk 2]
```diff
@@ -16,41 +16,53 @@
...
+        }
+
+    @task
+    def fetch_count(parameters):
+        http_hook = HttpHook(http_conn_id="cds", method="GET")
+        response = http_hook.run_with_advanced_retry(
+            endpoint=parameters["endpoint"],
+            _retry_args={
+                "stop": stop_after_attempt(3),
+                "retry": retry_if_exception_type(Exception),
```
Comment (drjova) [inline dags/library/cern_publication_records.py:35]: I would retry on request related exceptions, to avoid hiding underlying issues

**Review by drjova — CHANGES_REQUESTED** [review]
🙏 few comments

**Review by drjova — CHANGES_REQUESTED** [review]

File: dags/library/cern_publication_records.py [hunk 3]
```diff
@@ -16,41 +17,53 @@
...
-        data = request_again_if_failed(url=url, cds_token=cds_token)
-        total = get_total_results_count(data.text)
-        return {type_of_query: total}
+        url = get_endpoint(key, year)
+        return {
+            "endpoint": url,
+            "type_of_query": key,
+        }
+
+    @task
```
[the remark this answers is not in the document]
Comment (ErnestaP) [reply dags/library/cern_publication_records.py:29]: yes, we do! Adding it

File: dags/open_access/open_access.py [hunk 4]
```diff
@@ -15,47 +18,71 @@
...
         base_query = (
             r"(affiliation:CERN+or+595:'For+annual+report')"
             + rf"and+year:{year}+not+980:ConferencePaper+"
             + r"not+980:BookChapter"
         )
         type_of_query = [*query][0]
-        url = utils.get_url(query=f"{base_query}")
-        data = request_again_if_failed(url=url)
-        total = get_total_results_count(data.text)
+        query_p = rf"{base_query}+{query[type_of_query]}"
```
[the remark this answers is not in the document]
Comment (ErnestaP) [reply dags/open_access/open_access.py:31]: sure. query_p - p means that it's a query that goes in `p` parameter of the endpoint:
```
        return {
            "endpoint": rf"search?ln=en&cc={current_collection}&p={query_p}"
            + r"&action_search=Search&op1=a&m1=a&p1=&f1=&c="
            + r"Published+Articles&c=&sf=&so=d&rm=&rg=100&sc=0&of=xm",
            "type_of_query": type_of_query,
        }
```

File: tests/open_access/test_data_harvesting.py
Comment (drjova) [inline tests/open_access/test_data_harvesting.py:1]: this file is empty

File: dags/library/cern_publication_records.py [hunk 5]
```diff
@@ -8,10 +6,16 @@
     LibraryCernPublicationRecords,
 )
 from common.operators.sqlalchemy_operator import sqlalchemy_task
-from common.utils import get_total_results_count, request_again_if_failed
```
Comment (drjova) [inline dags/library/cern_publication_records.py:11]: can completely remove `request_again_if_failed`?

Comment (ErnestaP) [reply dags/library/cern_publication_records.py:11]: it is removed

**Review by drjova — APPROVED** [review]

**Pull request merged**